copy/multiple.go

```diff
@@ -140,7 +151,49 @@ func (c *copier) copyMultipleImages(ctx context.Context, policyContext *signatur
 				UpdateDigest:    updatedManifestDigest,
 				UpdateSize:      int64(len(updatedManifest)),
 				UpdateMediaType: updatedManifestType})
-		default:
+		}
+		if instance.op == instanceCopyClone {
+			processedOp = false
+			c.Printf("Replicating image %s (%d/%d)\n", instance.sourceDigest, i+1, len(instanceCopyList))
+			c.Printf("Replicating image %s (%d/%d)\n", instance.sourceDigest, i+1, len(instanceCopyList))
+			unparsedInstance := image.UnparsedInstance(c.rawSource, &instanceCopyList[i].sourceDigest)
+			src, err := image.FromUnparsedImage(ctx, options.SourceCtx, unparsedInstance)
+			if err != nil {
+				return nil, fmt.Errorf("initializing image from source %s: %w", transports.ImageName(c.rawSource.Reference()), err)
+			}
+			imgConfig, err := src.OCIConfig(ctx)
+			if err != nil {
+				return nil, fmt.Errorf("parsing image configuration: %w", err)
+			}
+			imgPlatform := imgspecv1.Platform{Architecture: imgConfig.Architecture,
```

Comment:
Why is this parsing the per-instance OCI config at all? We already have a platform specifier in the image we are cloning.

(And, in the wild, the index and config platforms do differ, with the config platform being incorrect because most consumers don’t inspect it. So using the config value is not harmless.)